lib/trino-orc/src/main/java/io/trino/orc/reader/SliceColumnReader.java

```diff
@@ -57,7 +58,7 @@ public SliceColumnReader(Type type, OrcColumn column, AggregatedMemoryContext sy
             throws OrcCorruptionException
     {
         requireNonNull(type, "type is null");
-        verifyStreamType(column, type, t -> t instanceof VarcharType || t instanceof CharType || t instanceof VarbinaryType);
+        verifyStreamType(column, type, t -> t instanceof VarcharType || t instanceof CharType || t instanceof VarbinaryType || t instanceof UuidType);
```

Comment:
UUID is not an ORC type. We should be reading from ORC using the native ORC type.

Reply:
i am not sure how i should follow your suggestion, please explain

Reply:
We should not touch the ORC library code here, as UUID is a Trino type. The ORC specification has a fixed list of types that ORC supports. UUID is not one of them. We should ask ORC to read as a binary.

Reply:
This should be listed as a supported type in ORC.  Until then, Iceberg and simply tell the orc reader it is binary.  Also, you will need to make sure you get the byte ordering correct for UUID, and if not you'll need to shuffle the ordering.  IIRC it changed in the last year or so.  @electrum do you remember this.

Oh ya, that reminds me.  UUID now uses an Int128ArrayBlock, so this is producing the wrong type of block.  This works but pollutes the callsites with different block types.

Reply:
i removed this, reading data as Varbinary. I don't know how to read varbinary and also read into Int128ArrayBlock at the same time, but it looks it's a performance improvement, so added a TODO to indicate this